src/charm.py

```diff
@@ -361,10 +211,13 @@ def _on_version_action(self, event):
 
         event.set_results(version)
 
-    def _on_node_configured_action(self, event):
-        """Remove node from DownNodes."""
-        # trigger reconfig
-        self._slurmd.configure_new_node()
+    def _on_node_configured_action(self, _: ActionEvent) -> None:
+        """Remove node from DownNodes and mark as active."""
+        # Trigger reconfiguration of slurmd node.
+        inv = self._slurmd.node_inventory
+        inv["new_node"] = False
+        self._slurmd.node_inventory = inv
```

Comment:
Is this line superfluous? inv is a reference to the self.slurmd.node_inventory dict, so updating "new_node" entry just updates the dict. I also wonder if this should actually go into the slurmd_interface rather than this action handler. As it is, it feels like the wrong abstraction to be editing data directly in that interface details rather than exposing an API for achieving the results.

Reply:
We can put this back into _interface_slurmd.py_ and expose it to `SlurmdCharm`, but I need to verify if updating the dict directly will cause the desired `RelationChangedEvent` to be emitted. `self._slurmd.node_inventory = ...` is invoking a property setter method that updates the slurmd integration data.

Reply:
Refactored in commit https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/16d137f544bf99c2fdfdfa462eef68c216e17769